pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -329,6 +347,149 @@ func (vm *VirtualMachine) GetVirtualSystemSettingData() (*VirtualSystemSettingDa
+		}
+
+		for _, wmiSyntheticNetworkAdapter := range wmiSyntheticNetworkAdapters {
+
+			syntheticNetworkAdapter, err := na.NewSyntheticNetworkAdapter(wmiSyntheticNetworkAdapter)
+			if err != nil {
+				continue
+			}
+
+			networkAdapterMacAddress, err := syntheticNetworkAdapter.GetPropertyAddress()
+			if err != nil {
+				continue
+			}
+
+			if strings.EqualFold(inputMacAddressHyperV, networkAdapterMacAddress) {
+				wmiGuestConfig, err := syntheticNetworkAdapter.GetRelated("Msvm_GuestNetworkAdapterConfiguration")
+				if err != nil {
+					return nil, err
+				}
+
+				wmiGuestConfigClone, err := wmiGuestConfig.Clone()
+				if err != nil {
+					return nil, err
+				}
+
+				return na.NewGuestNetworkAdapterConfiguration(wmiGuestConfigClone)
+			}
+		}
+	}
+
+	return nil, errors.Wrapf(err, "Unable to get the network adapter configuration for the vnic with mac address [%s] on vm [%s]", inputMacAddress, vm.ElementName)
+}
+
+func (vm *VirtualMachine) GetSecuritySettingData() (value *MsvmSecuritySettingData, err error) {
+	inst, err := vm.GetRelated("Msvm_Tpm")
+	if err != nil {
+		return nil, err
+	}
+	defer inst.Close()
+
+	// If the TPM is not found, then it is not configured or enabled
+	if inst == nil {
+		return nil, nil
+	}
+
+	tpmwmi, err := v2.NewMsvm_TPMEx1(inst)
+	if err != nil {
+		return nil, err
+	}
+
+	cimSettings, err := tpmwmi.GetRelatedSecuritySettingData()
+	if err != nil {
+		return nil, err
+	}
+
+	cimSettingsClone, err := cimSettings.Clone()
+	if err != nil {
+		return nil, err
+	}
+
+	securitySettings, err := v2.NewMsvm_SecuritySettingDataEx1(cimSettingsClone)
+	if err != nil {
+		return nil, err
+	}
+
+	return &MsvmSecuritySettingData{securitySettings}, nil
+}
+
+func (vm *VirtualMachine) GetOSConfiguration() (computerName string, isWindows bool, err error) {
+	inst, err := vm.GetRelated("Msvm_KvpExchangeComponent")
+	if err != nil {
+		return
+	}
+	defer inst.Close()
+
+	kvp, err := v2.NewMsvm_KvpExchangeComponentEx1(inst)
+	if err != nil {
+		return
+	}
+
+	guestIntrinsicItems, err := kvp.GetPropertyGuestIntrinsicExchangeItems()
+	if err != nil {
+		return
+	}
+
+	guestKvPairs := make(map[string]string)
+	for _, item := range guestIntrinsicItems {
+		var guestProperty INSTANCE
+		err = xml.Unmarshal([]byte(item), &guestProperty)
+		if err != nil {
+			return
+		}
+
+		var key, value string
+		for _, property := range guestProperty.PROPERTY {
+			switch property.NAME {
+			case "Name":
+				key = property.VALUE
+			case "Data":
+				value = property.VALUE
+			}
+		}
+
+		if key != "" {
+			guestKvPairs[key] = value
+		}
+	}
+
+	computerName, ok := guestKvPairs["FullyQualifiedDomainName"]
+	if !ok {
+		err = errors.Wrapf(errors.InvalidInput, "Unable to retrieve computer name via WMI")
+		return
+	}
+
+	osName, ok := guestKvPairs["OSName"]
+	if !ok {
+		err = errors.Wrapf(errors.InvalidInput, "Unable to retrieve operating system type via WMI")
+		return
+	}
+	if strings.Contains(strings.ToLower(osName), "window") {
```

Comment:
100% agree it makes sense to want to reduce the amount of user input for sure; I would just be concerned this makes hydration more error prone. 

I will say for my HCI example, that registry key does not contain 'window':
<img width="684" alt="image" src="https://github.com/user-attachments/assets/3ffec79e-f0c1-4188-8f49-8a779d9551ad" />

I assume a virtual HCI system running on HCI is not a common scenario or anything but am just highlighting that making this dependent on a public facing product name may not be the most future proof approach.

I won't hold up the PR over this since it's just a preview feature, but my vote for when this feature is GA would be to require the user to specify at least the OS type since we already make them do that for new VMs / gallery images.

Reply:
That is some great feedback! Thank you for uncovering a defect.

I can perform additional testing on different VM SKUs and probably update the logic while going for public preview.